File: Difficulty: Medium/Form the Largest Number/form-the-largest-number.cpp
class Solution {
  public:
 static bool cmp(string a,string b)
	{
	    return a+b>b+a;
	}
    string findLargest(vector<int> &a) {
        // code here
        int n=a.size();
        vector<string>arr;
        for(int i=0;i<n;i++)
        {
            arr.push_back(to_string(a[i]));
        }
        
        sort(arr.begin(),arr.end(),cmp);
        string ans="";
        for(int i=n-1;i>=0;i--)
        {   
            ans=arr[i]+ans;
        }
        string res="";
        int i=0;
        while(i<ans.size()-1 && ans[i]=='0')
        {
            i++;
        }
        res=ans.substr(i,ans.size()-i);
        return res;
        
    }
};

File: Difficulty: Medium/Form the Largest Number/form-the-largest-number.java
class Solution {
    public String findLargest(int[] arr) {
        // code here
         // Step 1: convert int[] to Integer[]
        Integer[] nums = new Integer[arr.length];
        for (int i = 0; i < arr.length; i++) {
            nums[i] = arr[i];
        }
        // Step 2: sort using Comparator
        Arrays.sort(nums, new Comparator<Integer>() {
            public int compare(Integer a, Integer b) {
                String x = a.toString();
                String y = b.toString();
                return (y + x).compareTo(x + y); // descending
            }
        });
          // Step 3: concatenate result
        StringBuilder ans = new StringBuilder();
        for (int i = 0; i < nums.length; i++) {
            ans.append(nums[i]);
        }

        // Step 4: handle leading zeros
        if (ans.charAt(0) == '0') {
            return "0";
        }

        return ans.toString();
        
        
    }
}